skfp/utils/splitters.py

```diff
@@ -0,0 +1,630 @@
+
+    train_subset: list[Any] = []
+    valid_subset: list[Any] = []
+    test_subset: list[Any] = []
+
+    if return_indices:
+        train_subset = train_ids
+        valid_subset = valid_ids
+        test_subset = test_ids
+    else:
+        train_subset = _get_data_from_indices(data, train_ids)
+        valid_subset = _get_data_from_indices(data, valid_ids)
+        test_subset = _get_data_from_indices(data, test_ids)
+
+    _check_subsets(train_subset, test_subset)
+
+    if len(valid_subset) == 0:
+        warnings.warn(
+            "Warning: Valid subset is empty. Consider using scaffold_train_test_split instead."
+        )
+
+    additional_data_split: list[Sequence[Any]] = []
+
+    additional_data_split = (
+        _split_additional_data(list(additional_data), [train_ids, valid_ids, test_ids])
+        if additional_data
+        else []
+    )
+
+    if additional_data:
+        return train_subset, valid_subset, test_subset, additional_data_split
+    else:
+        return train_subset, valid_subset, test_subset
+
+
+def _calculate_missing_sizes(
+    train_size: Optional[float],
+    valid_size: Optional[float],
+    test_size: Optional[float],
+) -> tuple[float, float, float]:
+    """
+    Calculate the missing sizes for train, validation, and test sets if they are not provided.
+
+    Parameters
+    ----------
+    train_size : float or None
+        The proportion of the data to use for training.
+    valid_size : float or None
+        The proportion of the data to use for validation.
+    test_size : float or None
+        The proportion of the data to use for testing.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the proportions for training, validation, and testing sets.
+    """
+    if train_size is None and test_size is None and valid_size is None:
+        return 0.8, 0.1, 0.1
+
+    if train_size is None and test_size is None:
+        raise ValueError("Either train_size or test_size must be provided")
+
+    if valid_size == 0.0:
+        warnings.warn(
+            "Validation set will not be returned since valid_size was set to 0.0."
+            "Consider using train_test_split instead."
+        )
+
+    if train_size is None:
+        if test_size is not None and valid_size is not None:
+            train_size = 1 - test_size - valid_size
+        else:
+            raise ValueError(
+                "Cannot determine train_size. Provide test_size and valid_size."
+            )
+
+    elif test_size is None:
+        if train_size is not None and valid_size is not None:
+            test_size = 1 - train_size - valid_size
+        else:
+            raise ValueError(
+                "Cannot determine test_size. Provide train_size and valid_size."
+            )
+
+    elif valid_size is None:
+        valid_size = 0.0
+        warnings.warn(
+            "valid_size set to 0, consider using scaffold_train_test_split() instead"
+        )
+
+    return train_size, test_size, valid_size
```

Comment:
We have quite complex logic here. Maybe simplify this and require user to either pass no of those 3 sizes (and use 80-10-10% then), or pass all three and just check that they sum up to 1, and print warning on zero-size validation size.